### PR TITLE
specify port range for the testing ignite instance

### PR DESCRIPTION
**Pull request by tyrasd**
This should allow unit tests to be run in parallel without interfering with each other. (hopefully :crossed_fingers: )

### CODE REVIEW

**Comment by joker234** [comment]
Is there a issue related to this PR? What does this PR try to solve? Parallel executions of `mvn test`? If yes, the solution wasn't sufficient, I still get Exceptions from the ignite test instance.

This are example errors I get when running in parallel in at least two shells in separate working directories:
```bash
% mvn test
…
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   TestMapReduceOSHDB_Ignite_ScanQuery.testOSMContributionView:6->TestMapReduce.testOSMContributionView:58 » ComputeUserUndeclared
[ERROR]   TestMapReduceOSHDB_Ignite_ScanQuery.testOSMContributionViewStream:6->TestMapReduce.testOSMContributionViewStream:123 » Cache
[ERROR]   TestMapReduceOSHDB_Ignite_ScanQuery.testOSMEntitySnapshotView:6->TestMapReduce.testOSMEntitySnapshotView:91 » ComputeUserUndeclared
[ERROR]   TestMapReduceOSHDB_Ignite_ScanQuery.testOSMEntitySnapshotViewStream:6->TestMapReduce.testOSMEntitySnapshotViewStream:160 » Cache
[ERROR]   TestMapReduceOSHDB_Ignite_ScanQuery.testTimeoutMapReduce »  Unexpected excepti...
[INFO] 
[ERROR] Tests run: 125, Failures: 0, Errors: 5, Skipped: 0
…
```

**Comment by tyrasd** [comment]
> Parallel executions of mvn test?

yes, that was the rationale behind it.

**Comment by tyrasd** [comment]
> errors I get when running in parallel in at least two shells in separate working directories

confirming. I thought these were gone in my testing, but maybe I didn't test it properly.

The first actual exception I get is:

```
class org.apache.ignite.IgniteCheckedException: Failed to serialize object: org.heigit.bigspatialdata.oshdb.api.mapreducer.backend.OSHDBIgniteMapReduceComputeTask$1@[…]
[…]
Caused by: java.io.NotSerializableException: org.heigit.bigspatialdata.oshdb.index.XYGridTree$3
```

:thinking:

**Comment by tyrasd** [comment]
ok, I guess the problem was that the _ipFinder_ property still had all possible ports listed (`<value>127.0.0.1:47577..47599</value>`), so the test client would always connect to the first test-server even when there were multiple ones running.

This should be addressed "properly" now (choosing a 1 in 1000 random port for each test run).

File: oshdb-api/src/test/java/org/heigit/bigspatialdata/oshdb/api/tests/TestMapReduceOSHDB_Ignite.java [hunk 1]
```diff
@@ -1,28 +1,50 @@
...
 import org.heigit.bigspatialdata.oshdb.util.CellId;
 import org.heigit.bigspatialdata.oshdb.util.TableNames;
-import static org.junit.Assert.fail;
 
 abstract class TestMapReduceOSHDB_Ignite extends TestMapReduce {
-  final static Ignite ignite =
-      Ignition.start(new File("./src/test/resources/ignite-config.xml").toString());
+  final static Ignite ignite;
+  static {
+    int rndPort = 47577 + (int) (Math.random() * 1000);
```
Comment (joker234) [inline oshdb-api/src/test/java/org/heigit/bigspatialdata/oshdb/api/tests/TestMapReduceOSHDB_Ignite.java:34]: Is this range supposed to be that large?

**Review by joker234 — APPROVED** [review]

File: oshdb-api/src/test/java/org/heigit/bigspatialdata/oshdb/api/tests/TestMapReduceOSHDB_Ignite.java (hunk 1 above)
Comment (tyrasd) [reply oshdb-api/src/test/java/org/heigit/bigspatialdata/oshdb/api/tests/TestMapReduceOSHDB_Ignite.java:34]: yeah, in order to make random "collisions" less likely.

**Pull request merged**